src/environment.js

```diff
@@ -122,9 +122,10 @@ if (process.env.NODE_ENV !== 'production') {
 
 export const defaultEthNode =
   getDefaultEthNode() || networkConfig.nodes.defaultEth
+
 export const web3Providers = {
-  default: new Web3.providers.WebsocketProvider(defaultEthNode),
-  wallet: window.web3 && window.web3.currentProvider,
+  default: provider(defaultEthNode),
+  wallet: provider(['injected', 'frame']),
 }
```

Comment:
⚠️ Ahhh also we're running into https://github.com/aragon/aragon/pull/469#discussion_r233074933; we can't detect if there's no wallet installed.

Perhaps we can use [`provider.connected`](https://github.com/floating/ethereum-provider/blob/master/index.js#L6) and assume that if you're not connected, you need to install an ethereum provider?

Reply:
I used `provider.status === 'connected'` instead. `provider.connected` exists on `ethereum-provider`, but if an injected provider is detected (e.g. MetaMask), it is used directly and MetaMask doesn’t set `provider.connected` :-(

https://github.com/aragon/aragon/pull/469/commits/0bc4b59f6d9ece6bda3620cc03a72875b30287b5#diff-485e8ff3c1ad0767039a519e7918034dR80